### ISSUE TITLE
Use FQDN for AssertionConsumerService Location in authentication request

### ISSUE THREAD

**Issue by nawatts** [fedushare/mech_saml_ec#23]
Currently, `gethostname` is used to build the GSS service name which is used as the Location for the [AssertionConsumerService](https://wiki.shibboleth.net/confluence/display/SHIB2/NativeSPAssertionConsumerService) in the authentication request. See [util_cred.c#L484](https://github.com/fedushare/mech_saml_ec/blob/c8d96a53babeb2851a02b3264b60693e63cc51fc/mech_saml_ec/util_cred.c#L484) and [SAML2XML.cpp#L278](https://github.com/fedushare/mech_saml_ec/blob/c8d96a53babeb2851a02b3264b60693e63cc51fc/mech_saml_ec/SAML2XML.cpp#L278).

However, on some machines, the hostname returned by `gethostname` does not match the external FQDN. For example, on the Palmetto cluster at Clemson, the hostname seen by `mech_saml_ec` is `user001` while the FQDN is `user.palmetto.clemson.edu`.

--- 

From [Kerberos documentation](https://web.mit.edu/kerberos/krb5-1.12/doc/appdev/gssapi.html):

> A GSSAPI application can name a local or remote entity by calling gss_import_name, specifying a name type and a value. The following name types are supported by the krb5 mechanism:

> * GSS_C_NT_HOSTBASED_SERVICE: The value should be a string of the form service or service@hostname

**Comment by jbasney** [comment]
Yeah, this is something we fixed in MyProxy but not in GSS-OpenSSH. OpenSSH needs a new GSSAPIHostName option that should override the gethostname call at [gss-serv.c#L112](https://github.com/gss-openssh/openssh-portable/blob/draft/gss-serv.c#L112). Then OpenSSH will pass that hostname in to ssh_gssapi_import_name and gss_acquire_cred to override the gethostname call in [util_cred.c#L484](https://github.com/fedushare/mech_saml_ec/blob/c8d96a53babeb2851a02b3264b60693e63cc51fc/mech_saml_ec/util_cred.c#L484). As a short-term fix, it seems reasonable to add a config option to override the gethostname call directly in [util_cred.c#L484](https://github.com/fedushare/mech_saml_ec/blob/c8d96a53babeb2851a02b3264b60693e63cc51fc/mech_saml_ec/util_cred.c#L484) when the admin wants a different GSS server identity from what is returned by gethostname.

**Comment by nawatts** [comment]
Where should this config option go? Environment variable? A `mech_saml_ec` specific config file under `/etc`?

**Comment by jbasney** [comment]
Another environment variable like MECH_SAML_EC_DEBUG and GSS_SAML_EC_CB_TYPE_FILE seems easiest but at some point mech_saml_ec probably does need a config file of its own (/etc/mech_saml_ec and ~/.mech_saml_ec).
